sklearn/preprocessing/data.py

```diff
@@ -1917,3 +1929,129 @@ def transform(self, X):
+
+    Apply boxcox transform on individual features with lambda
+    that maximizes the log-likelihood function for each feature
+
+    Parameters
+    ----------
+    X : array-like, shape [n_samples, n_features]
+        The data to be transformed. Should contain only positive data.
+
+    copy : boolean, optional, default is True
+        set to False to perform inplace transformation and avoid a
+        copy (if the input is already a numpy array or a scipy.sparse
+        CSR matrix and if axis is 1).
+
+    References
+    ----------
+    G.E.P. Box and D.R. Cox, "An Analysis of Transformations", Journal of the
+    Royal Statistical Society B, 26, 211-252 (1964).
+    """
+    X = check_array(X, ensure_2d=True, dtype=FLOAT_DTYPES, copy=copy)
+    if any(np.any(X <= 0, axis=0)):
+        raise ValueError("BoxCox transform can only be applied on positive data")
+    n_features = X.shape[1]
+    outputs = Parallel(n_jobs=-1)(delayed(_boxcox)(X[:, i], None) for i in range(n_features))
+    output = np.concatenate([o[..., np.newaxis] for o in outputs], axis=1)
+    return output
+
+
+class BoxCoxTransformer(BaseEstimator, TransformerMixin):
+    """BoxCox features individually.
+
+    Each feature (i.e. each column of the data matrix) will be applied
+    boxcox transform with lambda evaluated to maximise the log-likelihood
+
+    Parameters
+    ----------
+    feature_indices: "all" or array of indices or mask, default None
+        Specify what features are treated are to be transformed.
+
+        - 'all' (default): All features are to be transformed.
+        - array of indices: Array of feature indices to be transformed..
+        - mask: Array of length n_features and with dtype=bool.
+    Notes
+    -----
+    The Box-Cox transform is given by::
+
+        y = (x**lmbda - 1) / lmbda,  for lmbda > 0
+            log(x),                  for lmbda = 0
+
+    `boxcox` requires the input data to be positive.
+
+    This estimator is stateless (besides constructor parameters), the
+    fit method does nothing but is useful when used in a pipeline.
+
+    See also
+    --------
+    boxcox_transform: Equivalent function without the object oriented API.
+    """
+
+    def __init__(self, feature_indices=None, copy=False):
+        self.feature_indices = feature_indices
+        self.copy = copy
+
+    def fit(self, X, y=None):
+        """Do nothing and return the estimator unchanged
+
+        Parameters
+        ----------
+        X : array-like, shape [n_samples, n_features]
+            The data to fit by apply boxcox transform,
+            to each of the features and learn the lambda.
+        """
+        X = check_array(X, ensure_2d=True, dtype=FLOAT_DTYPES)
+        if any(np.any(X <= 0, axis=0)):
+            raise ValueError("BoxCox transform can only be applied on positive data")
+        n_features = X.shape[1]
+        outputs = Parallel(n_jobs=-1)(delayed(_boxcox)(X[:, i], None, True) for i in range(n_features))
```

Comment:
I was hoping Parallel would be better if it is applicable. I haven't timed it for an improvement. Shall I do that ? Also I thought feature_indices would be necessary for transform but during fit, it would be better to get the maximum likelihood parameters for all the features. I mean multiple calls for transform can be done after fitting the transformer if this is allowed. I will proceed as you suggest here.